### PR TITLE
W-16439901 asm eol gr

### PR DESCRIPTION
**Pull request by glenn-rodgers-sf**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

**Review by garciarodrigor — DISMISSED** [review]
LGTM 👍

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+        itemData:
+          message: "#['< Status=' ++ attributes.statusCode ++ ' Length=' ++ (attributes.headers['content-length'] default 'none') ++ ' Time=' ++ (attributes.headers['x-envoy-upstream-service-time'] default 'none')]"
+          level: "INFO"
+          secondSection: true
+    order: 0 
+EOF
+----
+
+[[create-a-service]]
+== Create a Service to Serve Traffic
```
Comment (garciarodrigor) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
== Create a Service to Service Traffic
```

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 2]
```diff
@@ -0,0 +1,540 @@
+= Integrating Flex Gateway with Istio Service Mesh
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:imagesdir: ../assets/images
+
+Integrate Flex Gateway into your Istio Service Mesh to leverage Flex Gateway policies, logging, and metrics from the Anypoint Platform control plane. Deploy Flex Gateway as an ingress gateway to manage incoming north-south cluster traffic and internal east-west traffic inside the cluster.
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: Consider putting the user value first. "To leverage...., integrate".

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+
+To deploy this Flex Gateway use case, use the following architecture diagram and tutorial. The use case is a modified version of the https://istio.io/latest/docs/examples/bookinfo/[Istio Bookinfo Application] with Flex Gateway integrated. In the use case:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: Instead of calling this the "use case", can you just say something like "integrate Flex Gateway with ...."?

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 4]
```diff
@@ -0,0 +1,540 @@
...
+
+* An Istio virtual service forwards all ingress north-south traffic to Flex Gateway.
+* Direct access to the `bookinfo` name space is restricted. All external traffic into the `bookinfo` namespace must come from the `flex` namespace. Istio automatically configures mTLS configuration between the `bookinfo` and `flex`.
+* Flex Gateway protects and monitors the exposed product page service from incoming traffic with an API instance.
+* The Istio sidecar intercepts east-west traffic between the product page service and the reviews service. Flex Gateway acts as an ingress gateway managing incoming traffic to the reviews service.
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc:14]: "reviews service" is correct?

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 5]
```diff
@@ -0,0 +1,540 @@
...
+
+image:flex-istio-deployment.png["A detailed image of the previously described use case."]
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: The alt text should be a sentence fragment that describes the image, but not be self-referential. (Don't say "A detailed image")

Can you try writer.ai to generate alt text?

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 6]
```diff
@@ -0,0 +1,540 @@
...
+
+NOTE: In the diagram, Flex Gateway is running in Connected Mode. If Flex gateway was running is local mode, the Anypoint Platform control plane is not present.
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
NOTE: In the diagram, Flex Gateway is running in Connected Mode. If Flex Gateway runs in local mode, the Anypoint Platform control plane is absent.
```

Comment (hannanelson) [reply gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: Connected Mode is init cap, but local mode is not. Correct?

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 7]
```diff
@@ -0,0 +1,540 @@
...
+
+To configure the architecture:
+
+. Review <<before-you-begin>>.
+. <<setup-environment>>.
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
. <<setup-environment, Set up your environment>>.
```

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 8]
```diff
@@ -0,0 +1,540 @@
...
+. <<forward-ingress-traffic>>.
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: Can you make the link text sentence case?

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 9]
```diff
@@ -0,0 +1,540 @@
...
+. <<create-a-service>>.
+
+[[before-you-begin]]
+== Before You Begin
+
+Before you begin, install the following prerequisites:
+
+* https://docs.docker.com/get-docker/[Docker^]
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: (no need to add the ^ - added automatically now)

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 10]
```diff
@@ -0,0 +1,540 @@
...
+
+* xref:flex-install.adoc#kubernetes[Download Flex Gateway for Use in a Kubernetes Cluster]
+
+* https://k3d.io/v5.0.0/#installation[K3d] to create a kubernetes cluster inside of docker
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: Should kubernetes be init cap?

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 11]
```diff
@@ -0,0 +1,540 @@
...
+
+* https://kubernetes.io/docs/tasks/tools/#kubectl[kubectl] to control the Kubernetes cluster
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc:38]: or lowercase?

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 12]
```diff
@@ -0,0 +1,540 @@
...
+
+* https://helm.sh/docs/intro/install/[Helm] to install Flex Gateway and Istio
+
+[[setup-environment]]
+== Setup Environment
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
== Set up Environment
```

Comment (hannanelson) [reply gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: set up = verb
setup = noun/adj

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 13]
```diff
@@ -0,0 +1,540 @@
...
+[source,ssh]
+----
+helm repo add istio https://istio-release.storage.googleapis.com/charts \
+&& helm repo add flex-gateway https://flex-packages.anypoint.mulesoft.com/helm \
+&& helm repo up
+----
+
+. Install Istio:
++
+.. Install the Istio base chart that contains the cluster-wide Custom Resource Definitions (CRDs) that must be installed prior to the deployment of the Istio control plane:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
.. Install the Istio base chart that contains the cluster-wide Custom Resource Definitions (CRDs) that must be installed before deploying the Istio control plane:
```

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 14]
```diff
@@ -0,0 +1,540 @@
...
+.. Install an Istio ingress gateway:
++
+[source,ssh]
+----
+helm upgrade -i --wait --create-namespace -n istio-system istio-ingress istio/gateway --version 1.21.2
+----
+
+. Install Flex Gateway:
+
+.. Create the namespace and setup the label to enable the sidecar injection:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
.. Create the namespace and set up the label to enable the sidecar injection:
```

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 15]
```diff
@@ -0,0 +1,540 @@
...
++
+[source,ssh]
+----
+kubectl -n test get pods
+----
+
+[[forward-ingress-traffic]]
+== Forward Ingress Traffic
+
+.  Create the Flex Gateway ingress API instance to forward all north-south incoming traffic in the `bookinfo` namespace to the `productpage` service:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: Does this work with the extra space?

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 16]
```diff
@@ -0,0 +1,540 @@
...
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:imagesdir: ../assets/images
+
+Integrate Flex Gateway into your Istio Service Mesh to leverage Flex Gateway policies, logging, and metrics from the Anypoint Platform control plane. Deploy Flex Gateway as an ingress gateway to manage incoming north-south cluster traffic and internal east-west traffic inside the cluster.
+
+To deploy this Flex Gateway use case, use the following architecture diagram and tutorial. The use case is a modified version of the https://istio.io/latest/docs/examples/bookinfo/[Istio Bookinfo Application] with Flex Gateway integrated. In the use case:
+
+* An Istio virtual service forwards all ingress north-south traffic to Flex Gateway.
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc:11]: Previously (and subsequently), you use incoming rather than ingress. Suggest making consistent.

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc (hunk 15 above)
Comment (hannanelson) [reply gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: I think incoming should come before north-south.

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 17]
```diff
@@ -0,0 +1,540 @@
...
+----
+kubectl -n test get pods
+----
+
+[[forward-ingress-traffic]]
+== Forward Ingress Traffic
+
+.  Create the Flex Gateway ingress API instance to forward all north-south incoming traffic in the `bookinfo` namespace to the `productpage` service:
++
+** Local Mode:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc:158]: init cap here... should it always be init cap?

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 18]
```diff
@@ -0,0 +1,540 @@
...
+*** *Protocol*: *HTTP*
+*** *Port*: `8081`
+*** *Base Path*: `/ingress/`
++
+Configure the API instance with the following upstream parameters:
++
+*** *Route label*: `productpage`
+*** *Upstream URL*: `http://productpage.bookinfo.svc:9080`
+
+. Create the `flex` Istio ingress gateway to forward the ingress traffic to Flex Gateway using a istio virtual service:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: Should istio be init cap here?

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 19]
```diff
@@ -0,0 +1,540 @@
...
+    - destination:
+        host: flex
+        port:
+          number: 8081
+    rewrite:
+      uri: /ingress/
+EOF
+----
++
+.. Wait a few seconds (approximately 10) and send a test request the `/api/v1/products/1` page:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
.. Wait a few seconds (approximately 10) and send a test request to the `/api/v1/products/1` page:
```

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 20]
```diff
@@ -0,0 +1,540 @@
...
++
+[source,ssh]
+----
+curl -v http://localhost/api/v1/products/1
+----
++
+.. Ensure you receive the following output:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
.. Ensure you receive this output:
```

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 21]
```diff
@@ -0,0 +1,540 @@
...
+< date: Tue, 27 Feb 2024 22:06:33 GMT
+< content-type: application/json
+< content-length: 195
+< x-envoy-upstream-service-time: 8
+<
+* Connection #0 to host localhost left intact
+{"id": 2, "author": "William Shakespeare", "year": 1595, "type": "paperback", "pages": 200, "publisher": "PublisherA", "language": "English", "ISBN-10": "1234567890", "ISBN-13": "123-1234567890"}
+----
+
+. Protect the Flex Gateway API instance with the Basic Authentication policy applied:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: "with the" -> "by applying the"?

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 22]
```diff
@@ -0,0 +1,540 @@
...
+----
++
+.. Wait a few seconds (approximately 10) and send a test request:
++
+[source,ssh]
+----
+curl -v http://localhost/
+----
++
+.. Ensure you receive a `401` response.
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
.. Make sure you receive a `401` response.
```

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 23]
```diff
@@ -0,0 +1,540 @@
...
+        itemData:
+          message: "#['< Status=' ++ attributes.statusCode ++ ' Length=' ++ (attributes.headers['content-length'] default 'none') ++ ' Time=' ++ (attributes.headers['x-envoy-upstream-service-time'] default 'none')]"
+          level: "INFO"
+          secondSection: true
+    order: 0 
+EOF
+----
+
+[[create-a-service]]
+== Create a Service to Service Traffic
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc:332]: Should this be "Service-to-Service"?

Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: Is it "Service-to-Service" or is it a service that services traffic? This is hard to parse.

Comment (hannanelson) [reply gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: Should it be "to Forward Traffic"?

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 24]
```diff
@@ -0,0 +1,540 @@
...
+          level: "INFO"
+          secondSection: true
+EOF
+----
+
+** Connected Mode:
++
+To add an API instance in Connected Mode, see xref:api-manager::create-instance-task-flex.adoc[].
++
+Configure the API instance with the following downstream parameters:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
Configure the API instance with these downstream parameters:
```

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 25]
```diff
@@ -0,0 +1,540 @@
...
++
+*** *Protocol*: *HTTP*
+*** *Port*: `8081`
+*** *Base Path*: `/bookinfo/reviews/`
++
+Configure the API instance with the following upstream parameters:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
Configure the API instance with these upstream parameters:
```

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 26]
```diff
@@ -0,0 +1,540 @@
...
++
+*** *Route label*: `reviews`
+*** *Upstream URL*: `http://reviews.bookinfo.svc:9080`
+
+
+. Intercept the traffic between the `productpage` service to the `reviews` service and forward it to `flex` service:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
. Intercept the traffic between the `productpage` service and the `reviews` service and forward it to the `flex` service:
```

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 27]
```diff
@@ -0,0 +1,540 @@
...
+    - destination:
+        host: flex.flex.svc.cluster.local
+        port:
+          number: 8081
+    rewrite:
+      uri: /bookinfo/reviews/
+EOF
+----
++
+.. Send a test request with the `foo` user and `bar` password to the `/productpage`:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: Avoid [foobar](https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-%22foo,foobar,fubar,FUBAR%22...):

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 28]
```diff
@@ -0,0 +1,540 @@
...
++
+[source,ssh]
+----
+curl -v -u foo:bar http://localhost/productpage
+----
++
+.. Check the Flex Gateway logs to ensure the request sent from the `productpage` service to the `reviews` service passed through the Flex Gateway. The logs appear similar to the following:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
.. Check the Flex Gateway logs to ensure the request sent from the `productpage` service to the `reviews` service passed through Flex Gateway. The logs appear similar to:
```

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 29]
```diff
@@ -0,0 +1,540 @@
...
+----
++
+.. Send a test request from the test namespace to the `details` service:
++
+[source,ssh]
+----
+kubectl -n test exec -ti -c sleep $(kubectl -n test get pod --no-headers -o=name) -- curl -v details.bookinfo:9080/details/0
+----
++
+.. Ensure you receive the following `403 forbidden` response:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
.. Make sure you receive this `403 forbidden` response:
```

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 30]
```diff
@@ -0,0 +1,540 @@
...
+< content-type: text/plain
+< date: Thu, 09 May 2024 15:51:11 GMT
+< server: envoy
+< x-envoy-upstream-service-time: 0
+<
+* Connection #0 to host details.bookinfo left intact
+RBAC: access denied
+----
++
+.. Send the following request to the `details` service to find how to access the `bookinfo` service from the `flex` namespace:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
.. Send this request to the `details` service to find out how to access the `bookinfo` service from the `flex` namespace:
```

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc [hunk 31]
```diff
@@ -0,0 +1,540 @@
...
++
+[source,ssh]
+----
+kubectl -n flex exec -ti -c app  $(kubectl -n flex get pod --no-headers -o=name) -- flexctl check http http://details.bookinfo:9080/details/0
+----
++
+.. Ensure the following response:
```
Comment (hannanelson) [inline gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]:
```suggestion
.. Make sure you receive this response:
```

File: gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc (hunk 23 above)
Comment (glenn-rodgers-sf) [reply gateway/1.8/modules/ROOT/pages/flex-gateway-istio.adoc]: I agree, it was an engineering suggestion, let me see if we can change it.

**Review by hannanelson — APPROVED** [review]
